lib/cocoapods/installer/analyzer.rb

```diff
@@ -702,7 +708,8 @@ def dependencies_for_specs(specs, platform, all_specs)
       # @return [PodTarget]
       #
       def generate_pod_target(target_definitions, target_inspections, specs, scope_suffix: nil, build_type: nil)
-        target_requires_64_bit = target_definitions.all? { |td| requires_64_bit_archs?(td.platform) }
+        object_version = target_inspections.values.map { |ti| ti.project.object_version }.min
```

Comment:
I guess a project can have multiple object versions? Maybe due to using different `.xcodeproj` I guess?

Reply:
if so is this handled from here? It seems like it is

Reply:
my thought was if the targets are using different user projects then they _could_ be different. super edge case, but yeah I picked the min version in that scenario